Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2250,6 +2253,30 @@ public void onActivityDestroyed(Activity activity) {
 
     }
 
+    private void startSession(Activity activity) {
+        Uri intentData = null;
+        if (activity.getIntent() != null) {
+            intentData = activity.getIntent().getData();
+        }
+        initSessionWithData(intentData, activity); // indicate  starting of session.
+    }
+
+    /*
+     * Check for forced session restart. The Branch session is restarted if the incoming intent has $branch_force_new_session set to true.
+     * This is for supporting opening a deep link path while app is already running in the foreground. Such as clicking push notification while app in foreground.
+     *
+     */
+    private boolean checkIntentForSessionRestart(Intent intent) {
+        boolean isRestartSessionRequested = false;
+        if (intent != null) {
+            isRestartSessionRequested = intent.getBooleanExtra(Defines.Jsonkey.ForceNewBranchSession.getKey(), false);
```

Comment:
If the use cloud push, can they still modify the intent parameters? Right now, we tell devs to put the key "branch" in the payload of their push, and the link URL as the value. Is this an additional parameter they'd need to add to the payload?